node/resolvers/catalog/product.ts

```diff
@@ -57,6 +57,15 @@ export const resolvers = {
 
     recommendations: product => product,
 
-    titleTag: ({productTitle}) => productTitle,
+    titleTag: ({ productTitle }) => productTitle,
+
+    specificationGroups: product => {
+      const buildPropertie = nameGroup => map((name: string) => ({ name, values: product[name] }), product[nameGroup] || [])
+      const allSpecificationsGroups = product.allSpecificationsGroups.concat(["allSpecifications"])
+      const specificationGroups = map((name: string) => ({ name, specifications: buildPropertie(name) }), allSpecificationsGroups || [])
```

Comment:
Do you really need that helper function?

I think it might be easier to read if you keep the code more linear, something like:

```js
  const allSpecificationsGroups = product.allSpecificationsGroups.concat(["allSpecifications"])

  const specificationGroups = allSpecificationsGroups.map(
    (groupName: string) => ({
      name: groupName,
      specifications: map(
        (name: string) => ({ name, values: product[name] }),
        product[groupName] || [],
      ),
    })
  )

  return specificationGroups
```

Maybe is because of the long lines lengths that is making things harder to read.